app_integrations/apps/salesforce.py

```diff
@@ -0,0 +1,395 @@
+
+    Salesforce will audit 5 types of event logs include 'Console', 'Login',
+    'LoginAs', 'Report' and 'ReportExport' for user access audit and data loss
+    prevention, although Salesforce provides 42 types of event logs.
+
+    Console events:
+        contain information about the performance and use of Salesforce Consoles.
+
+    Login events:
+        contain details about your org's user login history.
+
+    LoginAs events:
+        contain details about what a Salesforce admin did while logged in as another user.
+
+    Report events:
+        contain information about what happened when a user ran a report and
+        captures report's size.
+
+    ReportExport:
+        events contain details about reports that a user exported.
+    """
+    _SALESFORCE_TOKEN_URL = 'https://login.salesforce.com/services/oauth2/token'
+    _SALESFORCE_QUERY_URL = ('{instance_url}/services/data/v{api_version}/'
+                             '{query}{start_time}{event_type}')
+    # Use the Query resource to retrieve log files.
+    _SALESFORCE_QUERY_FILTERS = ('query?q=SELECT+Id+,+EventType+,+LogFile+,+LogDate+,'
+                                 '+LogFileLength+FROM+EventLogFile+')
+
+    # Log files filters. These typs of events should be auditted for security purpose.
+    _SALESFORCE_EVENT_TYPES = {'Console', 'Login', 'LoginAs', 'Report', 'ReportExport'}
+    # Use "Where" and "LogDate" clause to filter log files which are generated after
+    # last_timestamp.
+    _SALESFORCE_CREATE_AFTER = 'WHERE+LogDate+>+{}+'
+
+    _TIMEOUT = 15
+    EXCEPTIONS_TO_BACKOFF = (SalesforceAppError,)
+    BACKOFF_MAX_RETRIES = 3
+
+    def __init__(self, config):
+        super(SalesforceApp, self).__init__(config)
+        self._auth_headers = None
+        self._instance_url = None
+        self._latest_api_version = 0
+
+    @classmethod
+    def _type(cls):
+        return 'events'
+
+    @classmethod
+    def service(cls):
+        return 'salesforce'
+
+    @classmethod
+    def date_formatter(cls):
+        """Salesforce API date format: YYYY-MM-DDTHH:MM:SSZ"""
+        return '%Y-%m-%dT%H:%M:%SZ'
+
+    def _request_token(self):
+        """Request OAuth token frome salesforce
+
+        Meanwhile, it will also get instance url which will be used in futher
+        requests. The instance url identifies the Salesforce instance to which
+        API calls should be sent.
+
+        Returns:
+            bool: Returns True if update auth headers and instance url successfully.
+        """
+        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
+        # required credentials when request for a token.
+        data = {
+            'grant_type': 'password',
+            'client_id': self._config.auth['client_id'],
+            'client_secret': self._config.auth['client_secret'],
+            'username': self._config.auth['username'],
+            'password': self._config.auth['password']+self._config.auth['security_token'],
```

Comment:
please use str.format() for concatenation instead of `+` like: `'{}{}'.format(..., ...)`